### PR TITLE
Docs build tooling

### PR DESCRIPTION
**Pull request by jl-wynen**
Use the same CMake / Python scripts as in scipp

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -150,11 +150,17 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
...
-# Custom target for building documentation
+# Custom install target for docs to depend on.
 add_custom_target(
-  docs COMMAND ${Python_EXECUTABLE} ${CMAKE_SOURCE_DIR}/docs/build_docs.py
+  install-scippneutron COMMAND cmake --build ${CMAKE_CURRENT_BINARY_DIR}
+                               --target install
 )
 
+include(docs)
+add_docs_target(docs BUILDER html DEPENDS install-scippneutron)
```
Comment (OwenArnold) [inline CMakeLists.txt:160]: Why is it not possible to depend on the `install` target directly?

Comment (jl-wynen) [reply CMakeLists.txt:160]: You can only depend on targets that you defined yourself, not targets generated by CMake: https://cmake.org/cmake/help/latest/command/add_dependencies.html

Comment (OwenArnold) [reply CMakeLists.txt:160]: Ah yes. I think I've encountered this before. Thanks for reminding me.

File: CMakeLists.txt [hunk 2]
```diff
@@ -150,11 +150,17 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
...
+add_docs_target(doctest BUILDER doctest DEPENDS docs)
+add_docs_target(linkcheck BUILDER linkcheck DEPENDS docs)
```
Comment (OwenArnold) [inline CMakeLists.txt:162]: Missing `docs.cmake` (like scipp has), unknown function

Comment (jl-wynen) [reply CMakeLists.txt:162]: Oh, right

**Pull request merged**